### PR TITLE
feat: supports adding files inside the VM to support bundles

### PR DESCRIPTION
**Pull request by sam-berning**
Issue #, if available:

*Description of changes:*

Allows users to add files from inside their finch VM to a support bundle. The file can be specified in a config or in a command line flag with `vm:/path/to/file/in/vm`

*Testing done:*

```
make test-unit
```

manual testing

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/support/support.go [hunk 1]
```diff
@@ -108,7 +111,13 @@ func (bb *bundleBuilder) GenerateSupportBundle(additionalFiles []string, exclude
...
 			continue
 		}
-		err := bb.copyInFile(writer, file, path.Join(zipPrefix, logPrefix))
+		bb.logger.Debugf("Copying %s...", file)
+		var err error
+		if isFileFromVM(file) {
+			err = bb.streamFileFromVM(writer, file, path.Join(zipPrefix, logPrefix))
+		} else {
+			err = bb.copyInFile(writer, file, path.Join(zipPrefix, logPrefix))
+		}
```
Comment (ginglis13) [inline pkg/support/support.go:120]: nitpicky but could this be wrapped in a small util function, since this logic is repeated 2 other times?

**Review by ginglis13 — COMMENTED** [review]
LGTM, should we also add something to the helptext for `finch support-bundle generate` to document the use case of `vm:<file>`  ?

**Review by ginglis13 — DISMISSED** [review]
LGTM. Nonblocking IMO, but do you think this change warrants e2e tests as well?

**Comment by sam-berning** [comment]
> LGTM. Nonblocking IMO, but do you think this change warrants e2e tests as well?

Yeah, I'll make a follow up PR

File: pkg/support/support.go [hunk 2]
```diff
@@ -196,12 +214,50 @@ func (bb *bundleBuilder) copyInFile(writer *zip.Writer, fileName string, prefix
...
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+
+	waitStatus := make(chan error)
+	go func() {
+		err := cmd.Wait()
+		_ = pipeWriter.Close()
+		waitStatus <- err
```
Comment (mharwani) [inline pkg/support/support.go:244]: Does the error returned from `cmd.Wait()` display the correct error message, like "file not found"? Maybe we should read `errBuf` to generate the error message.

**Review by mharwani — APPROVED** [review]

**Review by ginglis13 — APPROVED** [review]

**Pull request merged**